### PR TITLE
Issue #15218: formatted input files for indentation sections to follow google style guide rules

### PR DESCRIPTION
**Pull request by Zopsss**
#15218 

Indentation module is used by 3 sections:

- [4.2 Block indentation: +2 spaces](https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.2-block-indentation)
- [4.5.2 Indent continuation lines at least +4 spaces](https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.5.2-line-wrapping-indent)
- [4.8.4.1 Indentation](https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.8.4.1-switch-indentation)

All these sections have their own input files directory but their input files contains same code i.e. they're duplicated.

@romani suggested to format all these sections in single PR only.

Currently, I have formatted the input files of section `4.2` only. I have found some false positives/negatives in input files, I have pointed those out. After discussing them, I will format the remaining sections

### CODE REVIEW

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class BlockIndentation2SpacesTest extends AbstractIndentationTestSupport
 
     @Override
     protected String getPackageLocation() {
-        return "com/google/checkstyle/test/chapter4formatting/rule4841indentation";
+        return "com/google/checkstyle/test/chapter4formatting/rule42blockindentation";
```
Comment (Zopsss) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java:34]: corrected the directory name

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java [hunk 2]
```diff
@@ -100,4 +100,9 @@ public void testCorrectAnnotationArrayInit() throws Exception {
         verifyWithConfigParser(MODULES, filePath);
     }
 
+    @Test
+    public void tetsFastMatcher() throws Exception {
+        final String filePath = getPath("InputFastMatcher.java");
+        verifyWithConfigParser(MODULES, filePath);
+    }
```
Comment (Zopsss) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java:107]: `InputFastMatcher.java` file was already present but was never tested, so created test for it

**Review by Zopsss — COMMENTED** [review]
Some extra changes:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectForAndParameter.java [hunk 3]
```diff
@@ -1,113 +1,193 @@
...
+                      "Loooooooooooooooooooooooooooooooo" // indent:19 exp:>=10
+                          + "oooooooooooooooooooooooooooooooooooooooo" // indent:21 exp:>=10
+                          + "oooooooooong"))) {} // indent:10 exp:10
+    } // indent:4 exp:4
+  } // indent:2 exp:2
+} // indent:0 exp:0
+
+class SecondForClassWithLongName1 // indent:0 exp:0
+implements Iterable<String> { // indent:8 exp:>=4
+  // violation above ''implements' has incorrect indentation level 0, expected level should be 4.'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectForAndParameter.java:171]: same as:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethods.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                "param1", // indent:4 exp:4
+                "param2")
+            .getInstance() // indent:32 exp:>=8
+      // violation below ''.' has incorrect indentation level 6, expected level should be 8.'
+      .doNothing("nothing")
+      .length();
+      // violation above ''.' has incorrect indentation level 6, expected level should be 8.'
+
+    int length2 = // indent:4 exp:4
+        // violation below ''new' has incorrect indentation level 4, expected level should be 8.'
```
Comment (Zopsss) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethods.java:25]: this file already contained violations, so we can ignore this.

Formatter corrected these indentations but I reverted them to as they were before.

**Review by Zopsss — COMMENTED** [review]
Some false positives/negatives I found. There were no violations before formatting the code. But after using the Formatter, inputs started giving some extra violations which I have pointed out here:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java [hunk 5]
```diff
@@ -1,113 +1,133 @@
...
-          : c) { //indent:10 exp:>=8
-      someFooMethod("longString", "veryLongString", //indent:6 exp:6
-          "superVeryExtraLongString"); //indent:10 exp:10
-    } //indent:4 exp:4
-  } //indent:2 exp:2
-} //indent:0 exp:0
+package com.google.checkstyle.test.chapter4formatting.rule42blockindentation; // indent:0 exp:0
+
+// indent:0 exp:0
+// indent:0 exp:0
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java:4]: just remove such lines.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java [hunk 6]
```diff
@@ -1,113 +1,133 @@
...
+  } // indent:2 exp:2
+
+  public void foo() { // indent:2 exp:2
+    int i = 0; // indent:4 exp:4
+    for (; i < 9; i++) { // indent:4 exp:4
+      if (veryLongLongLongCondition() // indent:6 exp:6
+          || veryLongLongLongCondition2()) { // indent:14 exp:>=10
+        someFooMethod(
+            "longString",
+            "veryLongString", // indent:8 exp:8
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java:35]: all comments  `// indent` should be removed.
as non of tools/test validate that numbers are valid in them.

this line is not 8 by identation.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java (hunk 1 above)
Comment (romani) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java:34]: can you keep old name? and create issue to change it after such a massive update in such files

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethods.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethods.java:25]: please put violation comment to be above `int length2 =`
comment in between of `=` and statement is weird edge case, lets not deal with it there.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethodsCorrect.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    doNothing( // indent:4 exp:4
+        someString // indent:8 exp:8
+            .concat(
+                "zyx" // indent:12 exp:12
+                ) // indent:12 exp:12
+            .concat(
+                "255, 254, 253" // indent:12 exp:12
+                ) // indent:12 exp:12
+        ); // indent:4 exp:4
+    // violation above ''method call rparen' has incorrect indentation level 8, expected level should be 4.'
```
[the remark this answers is not in the document]
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethodsCorrect.java]: false positive.
please create issue or find exiting. `);` is just line wrap, so it should be +4. So no violations is expected here.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectClass.java [hunk 8]
```diff
@@ -1,45 +1,48 @@
...
-          SecondClassReturnWithVeryVeryVeryLongName {  //indent:10 exp:>=6
-    public InnerClass2(String string) {  //indent:4 exp:4
-      super();  //indent:6 exp:6
+package com.google.checkstyle.test.chapter4formatting.rule42blockindentation; // indent:0 exp:0
+
+import java.util.Iterator; // indent:0 exp:0
+
+public class InputIndentationCorrectClass // indent:0 exp:0
+implements Runnable, Cloneable { // indent:4 exp:4
+// violation above ''implements' has incorrect indentation level 0, expected level should be 4.'
```
[the remark this answers is not in the document]
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectClass.java:7]: this is correct behavior, as this line is linewrap from `public class InputIndentationCorrectClass implements Runnable, Cloneable { `

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectClass.java [hunk 9]
```diff
@@ -1,45 +1,48 @@
...
+  } // indent:2 exp:2
+} // indent:0 exp:0
 
-  @Override  //indent:2 exp:2
-  public void run() {  //indent:2 exp:2
-    SecondClassWithLongLongLongLongName anon =  //indent:4 exp:4
-          new SecondClassWithLongLongLongLongName() {  //indent:10 exp:>=8
+class SecondClassWithLongLongLongLongName // indent:0 exp:0
+extends // indent:4 exp:4
+// violation above ''extends' has incorrect indentation level 0, expected level should be 4.'
```
[the remark this answers is not in the document]
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectClass.java:44]: nope, see above

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethodsCorrect.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/ClassWithChainedMethodsCorrect.java]: I guess it is not false positive. It is correct violation, I have opened an issue for it: https://github.com/checkstyle/checkstyle/issues/15336

> For line no 17 violation: `); // indent:4 exp:4 // warn`
>
> The closing bracket ( `);` ) is of method call `doNothing`, so the `);` should be indented same level as `doNothing`. This violation message is also correct IMO.

**Comment by rnveach** [comment]
Just to note,
https://github.com/checkstyle/checkstyle/compare/master...rnveach:checkstyle:indentation_replacer
Input files in 4th run can't be satisfied by our indentation check.

**Comment by Zopsss** [comment]
@romani removed trailing `// indent` comments and formatted the code. We loose edge cases after formatting and the false positives that were found at https://github.com/checkstyle/checkstyle/pull/15332#pullrequestreview-2190042702 are also gone now.

**Review by romani — APPROVED** [review]
Ok to merge.
Diversity is not good, but we can do it a bit later.

**Review by rdiachenko — APPROVED** [review]
The pr is too big to properly review it. I just skimmed it. I'm ok to merge it bacause it is mainly formatting.

**Pull request merged**